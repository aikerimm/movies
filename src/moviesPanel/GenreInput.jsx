import React from 'react';
import PropTypes from 'prop-types';

const GenreInput = ({genre, onGenreChanged, currentGenre}) => {
  return (
    <>
      <input
        id={`genre${genre}`}
        type='radio'
        checked={currentGenre === genre}
        onChange={() => onGenreChanged(genre)}
      />
      <label htmlFor={`genre${genre}`} className='genreRadioOption'>
        {genre}
      </label>
    </>
  );
};

GenreInput.propTypes = {
  genre: PropTypes.string.isRequired,
  onGenreChanged: PropTypes.func.isRequired,
  currentGenre: PropTypes.string.isRequired
};

export default React.memo(GenreInput);
