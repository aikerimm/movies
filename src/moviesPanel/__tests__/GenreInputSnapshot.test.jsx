import renderer from 'react-test-renderer';
import GenreInput from '../GenreInput';

it('GenreInput should render correctly', () => {
  const tree = renderer.create(<GenreInput genre='Comedy' onGenreChanged={jest.fn()} currentGenre='Horror'/>).toJSON();
  expect(tree).toMatchSnapshot();
});