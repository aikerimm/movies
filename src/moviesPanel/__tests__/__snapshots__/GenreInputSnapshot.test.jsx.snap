// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreInput should render correctly 1`] = `
[
  <input
    checked={false}
    id="genreComedy"
    onChange={[Function]}
    type="radio"
  />,
  <label
    className="genreRadioOption"
    htmlFor="genreComedy"
  >
    Comedy
  </label>,
]
`;
